testing/test_submit_scripts/test_submit_hf.py

```diff
@@ -0,0 +1,110 @@
+from shared_workflow.shared import set_wct as mocked_set_wct
+from shared_workflow.shared import resolve_header as mocked_resolve_header
+from estimation.estimate_wct import est_HF_chours_single as mocked_est_HF_chours_single
+from qcore.utils import load_yaml as mocked_load_yaml
+
+from testing.test_common_set_up import (
+    INPUT,
+    OUTPUT,
+    set_up,
+    get_bench_output,
+    get_input_params,
+    get_fault_from_rel,
+)
+
+import scripts.submit_hf
+
+
+def get_mocked_resolve_header(root_path, fault, realisation):
+    return lambda account, n_tasks, wallclock_limit, job_name, version, memory, exe_time, job_description, partition=None, additional_lines="", template_path="slurm_header.cfg", target_host=host, mail="test@test.com": mocked_resolve_header(
+        account,
+        n_tasks,
+        wallclock_limit,
+        job_name,
+        version,
+        memory,
+        exe_time,
+        job_description,
+        partition=partition,
+        additional_lines=additional_lines,
+        template_path=os.path.join(
+            os.path.basename(root_path),
+            "CSRoot",
+            "Runs",
+            fault,
+            realisation,
+            template_path,
+        ),
+        target_host=target_host,
+        mail=mail,
+    )
+
+
+def test_main(set_up, mocker):
+    """No return value. Just check that it runs without crashing"""
+    function = "main"
+
+    mocker.patch(
+        "scripts.submit_hf.set_wct", lambda x, y, z: mocked_set_wct(x, y, True)
+    )
+    mocker.patch("scripts.submit_hf.confirm", lambda x: False)
+
+    for root_path, realisation in set_up:
+        fault = get_fault_from_rel(realisation)
+        file_path = os.path.join(
+            root_path, INPUT, "{}_{}_{}.P".format("submit_hf.py", function, "args")
+        )
+        with open(file_path, "rb") as load_file:
+            args = pickle.load(load_file)
+
+        # Fault will probably change on each set of data, so reset these every time
+        mocker.patch(
+            "scripts.submit_hf.utils.load_sim_params",
+            lambda x: mocked_load_sim_params(
+                os.path.join(root_path, "CSRoot", "Runs", fault, realisation, x)
+            ),
+        )
+
+        mocker.patch(
+            "scripts.submit_hf.resolve_header",
+            get_mocked_resolve_header(root_path, fault, realisation),
+        )
+
+        mocker.patch(
+            "scripts.submit_hf.est.est_HF_chours_single",
+            lambda *args, **kwargs:
+            mocked_est_HF_chours_single(*args, *kwargs,
+                                        model_dir=os.path.join(root_path, 'AdditionalData', 'models', 'HF'))
+        )
+
+
+
+        scripts.submit_hf.main(args)
+
+
+def test_write_sl_script(set_up, mocker):
+    """The return value of write_sl_script is a filename that depends on the install location and current time and it
+    is there fore not practical to test it with """
```

Comment:
Why not at least check existence?

Reply:
Or does remove throw an error if file doesn't exist?

Reply:
os.remove throws an error if the file doesn't exist.

Reply:
Sweet. Maybe chuck a comment about that in the docstring.